flake.nix

```diff
@@ -0,0 +1,62 @@
+{
+  description = "WIP stuff to get started on the pinebook pro.";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs-channels/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    {
+      overlay = final: prev: import "${self}/overlay.nix" final prev;
+    } // flake-utils.lib.eachDefaultSystem (
```

Comment:
Do we need flake-utils here for only that?

Furthermore, it seems *incorrect* to me to assume we build for each default systems built on Hydra. It is tested, and expected to work only on `[  "x86_64-linux" "aarch64-linux" ]`.

Reply:
I guess this is similar to above. flake-utils just provides a few convenience functions like this that let you produce an attrset  from `system` with `packages`, `defaultPackage` etc and it makes it into an attrset with `package.<system>`, `defaultlPackage.<system>` etc. This one includes all the systems supported by NixOS, but there's another version that let's you pass a list of systems yourself so even if we want to limit it to those two, I think flake-utils is still useful. Of course though it's just a convenience if we want to avoid the dependency for some reason, though I'm not sure why we would.

Reply:
As far as platforms are concerned, let's not put things we haven't tried here, or else this sets expectations that we can't manage. If someone needs to build from Darwin, I don't think that having an untested attribute for the platform will help much, I would expect that they would make the changes required. Furthermore, I don't know the state of cross-compilation to a Linux platform from Darwin.

I guess the dependency's fine, though I'm not sure I like the outlook of flakes "needing" a trivial dependency like that for what seems to me like a trivial flake :/.

Reply:
Okay sounds good on platforms. As for flake-utils, I definitely see what yout saying. I think the fact that it isn't built-in somehow speaks to the immaturity of the flakes system, because I basically need this function in ever flake, even trivial ones. Something analagoua is implemented in the flakes for [nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/flake.nix#L15), [nix](https://github.com/NixOS/nix/blob/master/flake.nix#L18), etc. That being said it's very easy for me to just copy/reimplement the one function I use to avoid the dependency. Let me know which you want me to do.

Reply:
I really hesitate :). I don't care *that* much, but this does smell more and more like something that should be made available at a more basic layer. Since this is all locked-up in flake.lock, I guess it's fine to go with the external dep.

Reply:
Yeah I agree, it does smell like something that should be available at a more basic layer, and I guess I feel like depending on a single implementation of those utility functions will help get them into that basic layer instead of copy/pasting the same thing everywhere. So I'll leave it for now.